File: app/local_plugins/sts_polychat/android/src/main/kotlin/com/aiagent/sts_polychat/StsPolychatPlugin.kt
package com.aiagent.sts_polychat

import io.flutter.embedding.engine.plugins.FlutterPlugin
import com.aiagent.plugin_interface.NativeServiceRegistry
import com.aiagent.plugin_interface.VoitransWebRtcSession

/**
 * Flutter 插件入口：注册 vendor "polychat" 到 NativeServiceRegistry (STS)
 */
class StsPolychatPlugin : FlutterPlugin {

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        val context = binding.applicationContext
        NativeServiceRegistry.registerSts("polychat") { StsPolychatService(context) }
        // 预初始化 WebRTC Factory，避免首次连接卡顿
        VoitransWebRtcSession.warmup(context)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}

File: app/local_plugins/vendors/sts_polychat/ios/Classes/StsPolychatPlugin.swift
import Flutter
import ai_plugin_interface
import os.log

/// PolyChat STS Flutter plugin entry.
///
/// Mirrors the Android `StsPolychatPlugin`: registers vendor "polychat"
/// against `StsPolychatService` and pre-warms the shared WebRTC factory so
/// the first connection doesn't pay the init latency.
public class StsPolychatPlugin: NSObject, FlutterPlugin {
    private static let log = OSLog(subsystem: "com.aiagent.sts_polychat", category: "Plugin")

    public static func register(with registrar: FlutterPluginRegistrar) {
        NativeServiceRegistry.shared.registerSts("polychat") { StsPolychatService() }
        VoitransWebRtcSession.warmup()
        os_log("Registered NativeStsService for vendor: polychat",
               log: log, type: .debug)
    }
}
